### DISCUSSION TITLE
Support for custom tags in rpmspec

### DISCUSSION THREAD

**Discussion by huakim** [rpm-software-management/rpm#3307] [Ideas]
Is there any approach to create custom tags and sections? For example, they can be handled with lua functions

**Comment by pmatilai** [comment]
It's not a question of Lua or whatever. All information in rpm's is stored by pre-assigned tag numbers: https://rpm-software-management.github.io/rpm/manual/tags.html

Custom tags wont fit into that scheme, and we have no current plans to work on changing that.

**Reply by huakim** [reply]
> It's not a question of Lua or whatever. All information in rpm's is stored by pre-assigned tag numbers: https://rpm-software-management.github.io/rpm/manual/tags.html
> 
> Custom tags wont fit into that scheme, and we have no current plans to work on changing that.

No, i was not about rpm package file format, but was about rpm spec file.

**Reply by huakim** [reply]
> > It's not a question of Lua or whatever. All information in rpm's is stored by pre-assigned tag numbers: https://rpm-software-management.github.io/rpm/manual/tags.html
> > Custom tags wont fit into that scheme, and we have no current plans to work on changing that.
> 
> No, i was not about rpm package file format, but was about rpm spec file.

For now, we can use macros instead custom tags, for editing internal lua environment and printing additional tags during build rpm package, but custom tags would significally extend an usability of spec file. By the term of `tags` i means something like `Name: package`, where `Name` is tag. By the term of `custom tag`, i means something like `CustomTag: CustomProperty`, which can be used instead of writing `%CustomTag CustomProperty`. Of course, this would be good, if we also can somehow see values of `Requires`, `Provides`, `BuildRequires`  tags

**Reply by pmatilai** [reply]
Most spec tags are quite literally tags in the concrete package. 

It seems that you're asking to use that syntax for defining your own build-time variables. That's what macros are for.
Spec tags are reserved for rpm's own use. So sorry but no.
